### PR TITLE
Feature/firebase auth

### PR DESCRIPTION
**Pull request by florindakinandcarta**
implemented firebase authentication with email
you can register or login with email and password
profile fragment implemented gets the email of the user if logged in (other info like name etc. not available, to be implemented)

### CODE REVIEW

File: .idea/deploymentTargetDropDown.xml [hunk 1]
```diff
@@ -3,7 +3,20 @@
   <component name="deploymentTargetDropDown">
     <value>
       <entry key="app">
-        <State />
+        <State>
```
Comment (liridon-kin-and-carta) [inline .idea/deploymentTargetDropDown.xml:6]: All .idea files should be ignored and not pushed to the repository. Please check if you have .gitignore in order to ignore idea files and other files that are not needed. If not find how to setup .gitignore file.

File: app/src/main/java/com/example/erabook/firebaseActivities/LogIn.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    private lateinit var binding: ActivityLogInBinding
+    private lateinit var auth: FirebaseAuth
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityLogInBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        auth = Firebase.auth
+        binding.apply {
+            loginButton.setOnClickListener {
```
Comment (liridon-kin-and-carta) [inline app/src/main/java/com/example/erabook/firebaseActivities/LogIn.kt:26]: Please create a method called `setupOnClickListeners()` and then inside that method perform all the click listeners.

File: app/src/main/java/com/example/erabook/firebaseActivities/LogIn.kt [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                        baseContext,
+                        "Enter password please!",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+                auth.signInWithEmailAndPassword(emailLogIn.trim(), passwordLogIn)
+                    .addOnCompleteListener { task ->
+                        if (task.isSuccessful) {
+                            startActivity(Intent(this@LogIn, MainActivity::class.java).apply {
+                                putExtra("destination", R.id.userProfileFragment)
```
Comment (liridon-kin-and-carta) [inline app/src/main/java/com/example/erabook/firebaseActivities/LogIn.kt:47]: Please use constants for "destination"

File: app/src/main/java/com/example/erabook/firebaseActivities/LogIn.kt [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                            })
+                            Toast.makeText(
```
Comment (liridon-kin-and-carta) [inline app/src/main/java/com/example/erabook/firebaseActivities/LogIn.kt:49]: Same here create Toast extension if possible.

File: app/src/main/java/com/example/erabook/firebaseActivities/LogIn.kt [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+                                baseContext,
+                                "Log in successful.",
```
Comment (liridon-kin-and-carta) [inline app/src/main/java/com/example/erabook/firebaseActivities/LogIn.kt:51]: Same here, extract string resource.

File: app/src/main/java/com/example/erabook/firebaseActivities/LogIn.kt [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+                                Toast.LENGTH_SHORT,
+                            ).show()
+                        } else {
+                            Toast.makeText(
+                                baseContext,
+                                "Authentication failed.",
```
Comment (liridon-kin-and-carta) [inline app/src/main/java/com/example/erabook/firebaseActivities/LogIn.kt:57]: Same here, extract string resource.

File: app/src/main/java/com/example/erabook/firebaseActivities/LogIn.kt [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+                                Toast.LENGTH_SHORT,
+                            ).show()
+                        }
+                    }
+            }
+            goToRegister.setOnClickListener {
+                startActivity(Intent(baseContext, Register::class.java))
```
Comment (liridon-kin-and-carta) [inline app/src/main/java/com/example/erabook/firebaseActivities/LogIn.kt:64]: Please use "this" as context, or activityName.this, in order to reduce memory leaks.

File: app/src/main/java/com/example/erabook/firebaseActivities/LogIn.kt [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+            }
+            backLogin.setOnClickListener {
+                finish()
+                val intent = Intent(this@LogIn, MainActivity::class.java).apply {
+                    putExtra("destination", R.id.homeFragment)
```
Comment (liridon-kin-and-carta) [inline app/src/main/java/com/example/erabook/firebaseActivities/LogIn.kt:69]: Use constants please.

File: app/src/main/java/com/example/erabook/firebaseActivities/Register.kt [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+import android.view.View
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.widget.addTextChangedListener
+import com.example.erabook.databinding.ActivityRegisterBinding
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.ktx.Firebase
+
+class Register : AppCompatActivity() {
```
Comment (liridon-kin-and-carta) [inline app/src/main/java/com/example/erabook/firebaseActivities/Register.kt:14]: Rename to RegisterActivity please.

File: app/src/main/java/com/example/erabook/firebaseActivities/Register.kt [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+            confirmPasswordInput.editText?.addTextChangedListener { passwordEditable ->
+                if (passwordInput.editText?.text.toString().trim() != passwordEditable.toString()
+                        .trim()
+                ) {
+                    passwordMessage.visibility = View.VISIBLE
+                } else {
+                    passwordMessage.visibility = View.GONE
+                }
+            }
+            register.setOnClickListener {
```
Comment (liridon-kin-and-carta) [inline app/src/main/java/com/example/erabook/firebaseActivities/Register.kt:45]: Is "register" button? If so please rename to registerButton.

File: app/src/main/java/com/example/erabook/firebaseActivities/Register.kt [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+                        } else {
+                            Toast.makeText(
+                                baseContext,
+                                "Authentication failed.",
+                                Toast.LENGTH_SHORT,
+                            ).show()
+                        }
+                    }
+            }
+            alreadyHaveLogin.setOnClickListener {
```
Comment (liridon-kin-and-carta) [inline app/src/main/java/com/example/erabook/firebaseActivities/Register.kt:79]: Move onClickListeners to a separate method please.

File: app/src/main/java/com/example/erabook/firebaseActivities/Register.kt [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+                startActivity(Intent(baseContext, LogIn::class.java))
```
Comment (liridon-kin-and-carta) [inline app/src/main/java/com/example/erabook/firebaseActivities/Register.kt:80]: Use "this" as context please.

File: app/src/main/java/com/example/erabook/firebaseActivities/RegisterSuccessfulDialogFragment.kt [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+                container, false
+            )
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.apply {
+            OK.setOnClickListener {
```
Comment (liridon-kin-and-carta) [inline app/src/main/java/com/example/erabook/firebaseActivities/RegisterSuccessfulDialogFragment.kt:32]: What is "OK" please rename to a more meaningful name please.

File: app/src/main/java/com/example/erabook/firebaseActivities/RegisterSuccessfulDialogFragment.kt [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+                dialog?.dismiss()
+            }
+            goToLogin.setOnClickListener {
```
Comment (liridon-kin-and-carta) [inline app/src/main/java/com/example/erabook/firebaseActivities/RegisterSuccessfulDialogFragment.kt:35]: If goToLogin is a button then add suffix Button at the end please like: goToLoginButton.

File: app/src/main/java/com/example/erabook/firebaseActivities/RegisterSuccessfulDialogFragment.kt [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+                startActivity(Intent(requireContext(), LogIn::class.java))
+            }
+        }
+
```
Comment (liridon-kin-and-carta) [inline app/src/main/java/com/example/erabook/firebaseActivities/RegisterSuccessfulDialogFragment.kt:39]: Please delete this empty space.

File: app/src/main/res/layout/activity_log_in.xml [hunk 16]
```diff
@@ -0,0 +1,145 @@
...
+            android:background="@null"
+            android:contentDescription="@string/desc_img"
+            android:src="@drawable/arrow_back"
+            app:layout_constraintBottom_toTopOf="@id/hello_message"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <TextView
+            android:id="@+id/hello_message"
+            android:layout_width="match_parent"
```
Comment (liridon-kin-and-carta) [inline app/src/main/res/layout/activity_log_in.xml:25]: Please don't use match parent inside ConstraintLayout, refer to [documenation](https://developer.android.com/reference/androidx/constraintlayout/widget/ConstraintLayout)

File: app/src/main/res/layout/activity_log_in.xml [hunk 17]
```diff
@@ -0,0 +1,145 @@
...
+            android:text="@string/hello_message"
+            android:textSize="34sp"
+            android:textStyle="bold"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/backLogin" />
+
+        <TextView
+            android:id="@+id/signin_text"
+            android:layout_width="match_parent"
```
Comment (liridon-kin-and-carta) [inline app/src/main/res/layout/activity_log_in.xml:37]: Please don't use match parent inside ConstraintLayout, refer to [documenation](https://developer.android.com/reference/androidx/constraintlayout/widget/ConstraintLayout)

File: app/src/main/res/layout/activity_log_in.xml [hunk 18]
```diff
@@ -0,0 +1,145 @@
...
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <TextView
+            android:id="@+id/hello_message"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:padding="@dimen/space20"
+            android:text="@string/hello_message"
+            android:textSize="34sp"
```
Comment (liridon-kin-and-carta) [inline app/src/main/res/layout/activity_log_in.xml:29]: Extract text sizes to dimen folder please.

File: app/src/main/res/layout/activity_register.xml [hunk 19]
```diff
@@ -0,0 +1,180 @@
...
+        android:layout_margin="20dp"
+        android:background="@null"
+        android:contentDescription="@string/desc_img"
+        android:src="@drawable/arrow_back"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <TextView
+        android:id="@+id/acc_q"
+        android:layout_width="match_parent"
```
Comment (liridon-kin-and-carta) [inline app/src/main/res/layout/activity_register.xml:20]: Please don't use match parent inside ConstraintLayout, refer to [documenation](https://developer.android.com/reference/androidx/constraintlayout/widget/ConstraintLayout) instead use 0dp and set constraint properly

File: app/src/main/res/layout/activity_register.xml [hunk 20]
```diff
@@ -0,0 +1,180 @@
...
+        android:layout_height="wrap_content"
+        android:padding="20dp"
```
Comment (liridon-kin-and-carta) [inline app/src/main/res/layout/activity_register.xml:22]: Don't use hardcoded dp-s, extract it to dimen folder.

File: app/src/main/res/layout/activity_register.xml [hunk 21]
```diff
@@ -0,0 +1,180 @@
...
+        android:layout_height="34dp"
+        android:layout_margin="20dp"
+        android:background="@null"
+        android:contentDescription="@string/desc_img"
+        android:src="@drawable/arrow_back"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <TextView
+        android:id="@+id/acc_q"
```
Comment (liridon-kin-and-carta) [inline app/src/main/res/layout/activity_register.xml:19]: Please put a better naming here for this ID.

File: app/src/main/res/layout/activity_register.xml [hunk 22]
```diff
@@ -0,0 +1,180 @@
...
+        android:text="@string/already_login"
+        android:textColor="@color/primary"
+        android:textSize="14sp"
+        android:textStyle="bold"
+        app:layout_constraintBottom_toTopOf="@+id/register"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/remember_me" />
+    <Button
+        android:id="@+id/register"
```
Comment (liridon-kin-and-carta) [inline app/src/main/res/layout/activity_register.xml:168]: suggest: registerButton

File: app/src/main/res/layout/activity_register.xml [hunk 23]
```diff
@@ -0,0 +1,180 @@
...
+        android:layout_height="wrap_content"
+        android:layout_margin="20dp"
+        android:buttonTint="@color/primary"
+        android:text="@string/remember"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/confirm_password_input" />
+
+    <TextView
+        android:id="@+id/already_have_login"
```
Comment (liridon-kin-and-carta) [inline app/src/main/res/layout/activity_register.xml:154]: suggestion: alreadyHaveLoginTextView

File: app/src/main/res/layout/activity_register.xml [hunk 24]
```diff
@@ -0,0 +1,180 @@
...
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@string/password_match"
+        android:textColor="@color/bg_red"
+        android:textSize="14sp"
+        android:visibility="gone"
+        app:layout_constraintBottom_toTopOf="@id/confirm_password_input"
+        app:layout_constraintStart_toStartOf="@id/email_input" />
+    <CheckBox
+        android:id="@+id/remember_me"
```
Comment (liridon-kin-and-carta) [inline app/src/main/res/layout/activity_register.xml:143]: Suggestion: rememberMeCheckBox

File: app/src/main/res/layout/register_successful.xml [hunk 25]
```diff
@@ -0,0 +1,42 @@
...
+        android:padding="@dimen/space20"
+        android:text="@string/signup_succ_desc"
+        android:textSize="14sp"
+        android:textStyle="bold"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/signupsucc" />
+
+    <ImageView
+        android:id="@+id/OK"
```
Comment (liridon-kin-and-carta) [inline app/src/main/res/layout/register_successful.xml:32]: Raname this please.

**Review by liridon-kin-and-carta — CHANGES_REQUESTED** [review]
Good job 👍  I left some comments to clean it up a little bit.

File: .idea/deploymentTargetDropDown.xml (hunk 1 above)
Comment (florindakinandcarta) [reply .idea/deploymentTargetDropDown.xml:6]: I do have the .gitignore file in my project

File: app/src/main/res/layout/activity_log_in.xml (hunk 16 above)
Comment (florindakinandcarta) [reply app/src/main/res/layout/activity_log_in.xml:25]: fixed it in all the layouts :)

File: app/src/main/res/layout/activity_log_in.xml (hunk 18 above)
Comment (florindakinandcarta) [reply app/src/main/res/layout/activity_log_in.xml:29]: fixed it in all the layouts

File: app/src/main/java/com/example/erabook/util/Extensions.kt [hunk 26]
```diff
@@ -12,4 +14,8 @@ fun ImageView.loadImage(imageLink: String?) {
             .load(absolutePath)
             .into(this)
     }
+}
+
+fun Context.showToast(messageResId: Int, duration: Int = Toast.LENGTH_SHORT) {
```
Comment (liridon-kin-and-carta) [inline app/src/main/java/com/example/erabook/util/Extensions.kt:19]: Please annotate messageResId with @StringRes in order to accept only string resources and not normal Int values.

**Review by liridon-kin-and-carta — COMMENTED** [review]
Looks good, just one small comment for @StringRes annotation.

**Review by liridon-kin-and-carta — APPROVED** [review]
Great job, keep it up.

**Pull request merged**